src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -562,6 +565,9 @@ void Simulator::handle_message_hil_state_quaternion(const mavlink_message_t *msg
 		hil_lpos.vy = hil_state.vy / 100.0f;
 		hil_lpos.vz = hil_state.vz / 100.0f;
 		matrix::Eulerf euler = matrix::Quatf(hil_attitude.q);
+		hil_lpos.ax = acc(0);
+		hil_lpos.ay = acc(1);
+		hil_lpos.az = acc(2);
```

Comment:
```suggestion
	        matrix::Vector3f acc(hil_state.xacc / 1000.f, hil_state.yacc / 1000.f, hil_state.zacc / 1000.f);
	        acc = matrix::Quatf(hil_state.attitude_quaternion).conjugate(acc);
		hil_lpos.ax = acc(0);
		hil_lpos.ay = acc(1);
		hil_lpos.az = acc(2);
```

Reply:
> Could you move the acceleration calculation into the scope of where the local position is being broadcasted?

Seems OK, I'll test it again.

> It looks like you are converting the acceleration to body frame, while the topic lpos_ground_truth should be in the local frame

Yes, the original value is supposed to be in the local NED frame.
**But here comes the problem.** Without the quaternion conversion, the accel value provided by `gazebo` is far from the one in vehicle_lpos (the fused sensor state).

I tried multiple conversion rules, include keep or reverse az, use `quat.conjugate` or `quat.conjugate_inversed`, then compare the converted value against the sensor value in `vehicle_lpos`.
**The best rule is `keep az + quat.conjugate`.** Most value is close, but some large values are off from the sensor, possibly caused by sensor noise.

Reply:
@wangwwno1 I believe that is a bug from the gazebo side, since it is converting the values in the body frame: https://github.com/PX4/PX4-SITL_gazebo/blob/6fa6ec78a7a1619b7b90ac0c01aaec919defbe35/src/gazebo_mavlink_interface.cpp#L734

While handling the bug on gazebo side here will make the issue disappear, but I would rather fix the issue properly. Would you be able to fix the problem on the `PX4-SITL-gazebo` side also?

Reply:
> Would you be able to fix the problem on the PX4-SITL-gazebo side also?
> Is this also true for jmavsim?

I haven't test it in `jmavsim`, but I can do that after fix the conversion issue in `gazebo` side.
Maybe we should check whether `jmavsim` (and other simulator) acceleration follows the same local NED frame.

BTW, [this code line](https://github.com/mavlink/c_library_v2/blob/92b1a43468e8737da2d4cc1e72304e6443dcfdd3/common/mavlink_msg_hil_state_quaternion.h#L102) said the accel value should in the so-called `XYZ frame`.
Is the XYZ frame in `hil_state` equivalent to the local NED frame? If so, maybe we can make the comment more straight forward.

Reply:
Thanks @wangwwno1, that is a generated header file so probably we need to touch the mavlink message definitions itself